docs/design/security_context.md

```diff
@@ -0,0 +1,190 @@
+# Security Contexts
+## Abstract
+A security context is a set of constraints that are applied to a container in order to achieve the following goals (from [security design](security.md)):
+
+1.  Ensure a clear isolation between container and the underlying host it runs on
+2.  Limit the ability of the container to negatively impact the infrastructure or other containers
+
+## Background
+
+The problem of securing containers in Kubernetes has come up [before](https://github.com/GoogleCloudPlatform/kubernetes/issues/398) and the potential problems with container security are [well known](http://opensource.com/business/14/7/docker-security-selinux). Although it is not possible to completely isolate Docker containers from their hosts, new features like [user namespaces](https://github.com/docker/libcontainer/pull/304) make it possible to greatly reduce the attack surface.
+
+## Motivation
+
+### Container isolation
+
+In order to improve container isolation from host and other containers running on the host, containers should only be 
+granted the access they need to perform their work. To this end it should be possible to take advantage of Docker 
+features such as the ability to [add or remove capabilities](https://docs.docker.com/reference/run/#runtime-privilege-linux-capabilities-and-lxc-configuration) and [assign MCS labels](https://docs.docker.com/reference/run/#security-configuration) 
+to the container process.
+
+Support for user namespaces has recently been [merged](https://github.com/docker/libcontainer/pull/304) into Docker's libcontainer project and should soon surface in Docker itself. It will make it possible to assign a range of unprivileged uids and gids from the host to each container, improving the isolation between host and container and between containers.
+
+### External integration with shared storage
+In order to support external integration with shared storage, processes running in a Kubernetes cluster 
+should be able to be uniquely identified by their Unix UID, such that a chain of  ownership can be established. 
+Processes in pods will need to have consistent UID/GID/SELinux category labels in order to access shared disks.
```

Comment:
For disk, outside.  However, the user to run as inside the namespace may be something a user wants to change.  If namespaces are not present, a mismatch between the two should reject the pod, maybe.  


> On Feb 20, 2015, at 12:00 AM, Tim Hockin <notifications@github.com> wrote:
> 
> In docs/design/security_context.md:
> 
> > +
> > +## Motivation
> > +
> > +### Container isolation
> > +
> > +In order to improve container isolation from host and other containers running on the host, containers should only be 
> > +granted the access they need to perform their work. To this end it should be possible to take advantage of Docker 
> > +features such as the ability to [add or remove capabilities](https://docs.docker.com/reference/run/#runtime-privilege-linux-capabilities-and-lxc-configuration) and [assign MCS labels](https://docs.docker.com/reference/run/#security-configuration) 
> > +to the container process.
> > +
> > +Support for user namespaces has recently been [merged](https://github.com/docker/libcontainer/pull/304) into Docker's libcontainer project and should soon surface in Docker itself. It will make it possible to assign a range of unprivileged uids and gids from the host to each container, improving the isolation between host and container and between containers.
> > +
> > +### External integration with shared storage
> > +In order to support external integration with shared storage, processes running in a Kubernetes cluster 
> > +should be able to be uniquely identified by their Unix UID, such that a chain of  ownership can be established. 
> > +Processes in pods will need to have consistent UID/GID/SELinux category labels in order to access shared disks.
> Does this mean the in-namespace UID or the root-namespace UID?
> 
> —
> Reply to this email directly or view it on GitHub.
>